Lib/functools.py

```diff
@@ -447,54 +442,94 @@ class partialmethod:
     Supports wrapping existing descriptors and handles non-descriptor
     callables as instance methods.
     """
-    __new__ = _partial_new
+
+    __slots__ = ("func", "args", "keywords", "wrapper",
+                 "__dict__", "__weakref__")
+
     __repr__ = _partial_repr
 
-    def _make_unbound_method(self):
-        def _method(cls_or_self, /, *args, **keywords):
-            phcount = self._phcount
-            if phcount:
-                try:
-                    pto_args = self._merger(self.args + args)
-                    args = args[phcount:]
-                except IndexError:
-                    raise TypeError("missing positional arguments "
-                                    "in 'partialmethod' call; expected "
-                                    f"at least {phcount}, got {len(args)}")
-            else:
-                pto_args = self.args
-            keywords = {**self.keywords, **keywords}
-            return self.func(cls_or_self, *pto_args, *args, **keywords)
-        _method.__isabstractmethod__ = self.__isabstractmethod__
-        _method.__partialmethod__ = self
-        return _method
+    def __init__(self, func, /, *args, **keywords):
+        if isinstance(func, partialmethod):
+            # Subclass optimization
+            temp = partial(lambda: None, *func.args, **func.keywords)
+            temp = partial(temp, *args, **keywords)
+            func = func.func
+            args = temp.args
+            keywords = temp.keywords
+
+        self.func = func
+        self.args = args
+        self.keywords = keywords
+
+        if isinstance(func, _STD_METHOD_TYPES):
+            self.method = None
+        elif getattr(func, '__get__', None) is None:
+            if not callable(func):
+                raise TypeError(f'the first argument {func!r} must be a callable '
+                                'or a descriptor')
+            self.method = None
+        else:
+            # Unknown descriptor
+            self.method = _UNKNOWN_DESCRIPTOR
+
+    def _set_func_attrs(self, func):
+        func.__partialmethod__ = self
+        if self.__isabstractmethod__:
+            func = abstractmethod(func)
+        return func
+
+    def _make_method(self):
+        args = self.args
+        func = self.func
+
+        # 4 cases
+        if isinstance(func, staticmethod):
+            deco = staticmethod
+            method = partial(func.__wrapped__, *args, **self.keywords)
+        elif isinstance(func, classmethod):
+            deco = classmethod
+            ph_args = (Placeholder,) if args else ()
```

Comment:
This creates a new tuple on each invocation. Since Placeholder is a global and tuple is immutable, you could define a global:
```
single_placeholder = (Placeholder, )
```
and use that in improve construction performance (not sure it improves much).